### PR TITLE
Added response_type to serialized rulesets from flows API endpoint

### PR DESCRIPTION
**Pull request by rowanseymour**
Also tidied up json date functions to ease testing of such endpoints

### CODE REVIEW

File: temba/api/views.py [hunk 1]
```diff
@@ -2321,7 +2321,11 @@ class FlowEndpoint(generics.ListAPIView):
       * **archived** - whether this flow is archived (boolean) (filterable: ```archived```)
       * **labels** - the labels for this flow (string array) (filterable: ```label``` repeatable)
       * **created_on** - the datetime when this flow was created (datetime) (filterable: ```before``` and ```after```)
-      * **rulesets** - the rulesets on this flow, including their label, node id and integer id
+      * **rulesets** - the rulesets on this flow, including their node UUID label, and response type which is one of following string:
+
+             O - Open Ended
+             C - Multiple Choice
+             N - Numeric
```
Comment (norkans7) [inline temba/api/views.py]: more response types too

File: temba/values/models.py [hunk 2]
```diff
@@ -30,6 +31,7 @@
 # cache for up to 30 days (we will invalidate manually when dependencies change)
 VALUE_SUMMARY_CACHE_TIME = 60 * 60 * 24 * 30
 
+
```
Comment (rowanseymour) [inline temba/values/models.py]: Just being OCD and tidying up PEP whitespace violations....

**Comment by nicpottier** [comment]
This is a bit tricky.. Rulesets don't really have response types, responses have response types.

IE, you can have a ruleset that has both a "Has a number" and a "Has a date" as well as a "Has a string".

Values themselves have types yes, but rulesets I'm not so sure.

**Comment by rowanseymour** [comment]
I'm confused. We have `RuleSet.response_type` which is used by the editor to determine open-ended vs numeric vs multiple-choice etc right?. That's what I need access to in TracPro to determine correct chart type.

**Comment by nicpottier** [comment]
Yep, sorry, you are right. I'm ok with this, let's just not document it for the time being.

**Comment by ericnewcomer** [comment]
Yup.

**Pull request merged**